### PR TITLE
Not possible to send certain fractional amounts of tokens

### PR DESCRIPTION
**Pull request by buberdds**
Closes https://github.com/oasisprotocol/oasis-wallet-web/issues/533.

### CODE REVIEW

**Review by pro-wh — APPROVED** [review]
Good fix for that issue, thanks.

Small ideas for this PR:

- in the tests, group digits in the long big integers with `_`
- could we use Math.round() instead of toFixed() so that we don't have to take a trip through string type? should behave the same

The weakness of this is that we don't detect inputs that had too much precision which we coulnd't represent on chain, e.g. 5.1112223335. Ideally we'd still error out in those cases. But it's better to have this, and I'm in favor of merging it

**Comment by buberdds** [comment]
Thanks for your input. Can you give me sample code for # 1 as I am not following your idea about grouping digits yet. 

As for precision issue, again thanks for sharing that thought. It sounds more like a validation not parsing to me. I actually started tackling another issue related to amount field so I will look into validation as well.

File: src/app/lib/helpers.test.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { parseNumberToBigInt } from './helpers'
+
+describe('parseNumberToBigInt', () => {
+  it('should return a value of type bigint', () => {
+    expect(parseNumberToBigInt(33333.3)).toEqual(33333300000000n)
+    expect(parseNumberToBigInt(2.0121)).toEqual(2012100000n)
+    expect(parseNumberToBigInt(5)).toEqual(5000000000n)
```
Comment (pro-wh) [inline src/app/lib/helpers.test.ts:7]:
```suggestion
    expect(parseNumberToBigInt(33333.3)).toEqual(33333_300_000_000n)
    expect(parseNumberToBigInt(2.0121)).toEqual(2_012_100_000n)
    expect(parseNumberToBigInt(5)).toEqual(5_000_000_000n)
```

something like this, so it's easier to eyeball where 10^9 is.

File: src/app/lib/helpers.ts [hunk 2]
```diff
@@ -48,3 +48,5 @@ export function concat(...parts: Uint8Array[]) {
   }
   return result
 }
+
+export const parseNumberToBigInt = (value: number) => BigInt(Math.round(value * 10 ** 9))
```
Comment (lukaw3d) [inline src/app/lib/helpers.ts:52]: @pro-wh is this usually called converting to base units?

I'm not sure if that's accurate for `parseNumberToBigInt(action.payload.shares)`

Comment (pro-wh) [reply src/app/lib/helpers.ts:52]: Should it be something else? Lots of software, e.g. this and oasis scan, seem to be using giga shares.

Comment (lukaw3d) [reply src/app/lib/helpers.ts:52]: Not sure.

I rebased. Happy to merge

Comment (pro-wh) [reply src/app/lib/helpers.ts:52]: ok let's proceed to merge. there's been some talk of moving away from presenting shares or gigashares to end users completely, so we'll have a lot more design discussion when we do that

**Pull request merged**